shared/extension/behaviour/InformationSource.hpp

```diff
@@ -23,6 +23,33 @@
 #include <memory>
 #include <typeindex>
 
+namespace extension::behaviour {
+
+    /**
+     * Provides information about how and why this provider was executed.
+     */
+    struct RunInfo {
```

Comment:
Is it worth making this a smart enum and building in string conversions so you don't need the `decode_reason` function in your test?

Reply:
Seems a bit overkill just for a test